master/buildbot/reporters/message.py

```diff
@@ -16,21 +32,48 @@ class MessageFormatter(object):
     template_type = 'plain'
     wantProperties = True
     wantSteps = False
+    wantLogs = False
 
-    def __init__(self, template_name=None, template_dir=None, template_type=None):
+    def __init__(self, template_name=None, template_dir=None, template=None,
+                 subject_name=None, subject=None, template_type=None, ctx=None):
```

Comment:
subject_name is weird. probably this is a good time to change to template_filename and subject_filename (with a deprecation warning on template_name)